optuna/study.py

```diff
@@ -296,10 +305,12 @@ def optimize(
         if not self._optimize_lock.acquire(False):
             raise RuntimeError("Nested invocation of `Study.optimize` method isn't allowed.")
 
+        progress_bar = pbar_module.ProgressBar(
+            show_progress_bar or os.getenv('NO_PBAR') == 0, n_trials, timeout)
```

Comment:
Personally, I don't like to use environment variables as library API. Although it's a handy way to control the behavior, it feels a kind of a global variable and there is a risk to make it harder to debug and refactor.
I'm not sure, but I think that most of the Python lint tools cannot recognize and track environment variables, so refactoring the code could be difficult if the number of environment variables is increased. In addition, users might easily overlook a typo of the variable name.

Reply:
However, it's okay to accept this change, if the over dev members are positive to environment variables.

Reply:
Although I wouldn't want to bloat the `optimize` parameters, I basically share the same opinion about being cautious introducing environment variables, moreover, introducing multiple ways to control one same behavior (Read Zen of Python). I suggest discussing the introduction of environment variable controlled behavior outside this PR, and maybe put this PR on a hold until settled if deemed necessary. Another option would be to introduce this feature as experimental.

Reply:
OK, I now reckon that I should remove that environment variable. 

Just dropping the motivation of the environment variable. It was to make it easier to turn on/off the progress bar frequently during my initial implementation and it was really comfortable to me.

Reply:
I can definitely see that the environment variable is handy for implementing this feature, however, not sure that applies to our general user. 🙂